### ISSUE TITLE
CLJS Support

### ISSUE THREAD

**Issue by gtrak** [tomfaulhaber/autodoc#16]
Would you welcome CLJS support? 

I've got a working proof of concept in my branch, relying on a dumped compiler analysis file, the generation of which will potentially make it into cljs soon.  However, it could also be used with runtime-available data.

Screenshot:
http://i.imgur.com/BrAc61U.png?1

Diff:
https://github.com/gtrak/autodoc-cljs/commit/0c4bcf15345c8268ac83eb247e5c94da689fd47c

It's not quite ready for a proper pull-request, but I wanted to know if I should work towards a proper integration.

Here's a related issue I filed with lein-cljsbuild, which I hacked in order to generate the dump.
https://github.com/emezeske/lein-cljsbuild/issues/283

**Comment by venantius** [comment]
It's been quite a while since this issue was opened, but it would be great to see some movement on this. In particular, the ClojureScript project itself would love to have autodoc for generating documentation for its cljs.*.api namespaces (see http://dev.clojure.org/jira/browse/CLJS-1446). 

This will require support for both `.cljs` and `.cljc` files.

I'm not really sure how to go about approaching this as I ran into some unrelated issues when trying to build the current version of Autodoc, but I'll file a separate issue for that.

**Comment by tomfaulhaber** [comment]
I agree, but I'm not sure that @gtrak was on exactly the right track.

Here's how I would think about it:

1. Create a autodoc-collect variant that processes cljs files (and cljc files handled as clojurescript). It should produce an identical data structure to that produced by autodoc collect.
2. Teach autodoc to be smart about whether to call autodoc-collect, autodoc-collect-cljs, or both (the new tools.namespace should help a lot here).
3. Merge together the output of the two programs (especially interesting for cljc files).
4. Modify the output format to indicate language variant when two variants exist.

A slightly different approach would be to make autodoc-collect smart enough to handle both languages. This is dangerous, though, since autodoc-collect is specifically designed to handle clojure versions all the way back to v1.1 and is, therefore, pretty limited.

As I mentioned above, I am happy to have contributions in this area. One thing that would be a huge help work on the ClojureScript version of autodoc-collect. My expertise on the mechanics of ClojureScript is weak and not likely to get a lot better in the near future, so that would accelerate this issue a bunch.

**Comment by venantius** [comment]
I'm inclined to agree that a new approach may be needed. At the moment my familiarity with ClojureScript is still pretty weak as I only started tinkering with it a week or so ago, so I'm probably not the right person to take this on. If that changes I'll come back :P
